FirebaseRemoteConfig/Sources/RCNPersonalization.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "Interop/Analytics/Public/FIRAnalyticsInterop.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *const kAnalyticsOriginPersonalization = @"fp";
+static NSString *const kAnalyticsPullEvent = @"_fpc";
+static NSString *const kArmKey = @"_fpid";
+static NSString *const kArmValue = @"_fpct";
+static NSString *const kPersonalizationId = @"personalizationId";
+
+@interface RCNPersonalization : NSObject
+
+/// Analytics connector
+@property(nonatomic, strong) id<FIRAnalyticsInterop> _Nullable analytics;
+
+/// Returns the RCNPersonalization singleton.
++ (instancetype)sharedInstance;
```

Comment:
Excuse me in advance for the questions, I may not know all the design details, but I would like to clarify a couple things:

1. Why RCNPersonalization singleton is needed? Is it possible to use just an instance or static methods passing `FIRAnalyticsInterop`?
2. The singleton doesn't seem to respect `FIRApp` lifecycle. If a shared instance is actually required, then it is better to be implemented via components system - implement `FIRLibrary` protocol (see [Remote Config](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:FirebaseRemoteConfig/Sources/FIRRemoteConfigComponent.h;l=40;drc=d4621a636f0649c728d0f82819a6f307568afa0a) or simpler [FIS](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:FirebaseInstallations/Source/Library/FIRInstallations.m;l=54;drc=a3a59c1bfc37651837bab6f926b252eac2ed2343) examples, etc.)

Reply:
It doesn't really need to be singleton, so just made it an instance.